hips/utils/wcs.py

```diff
@@ -0,0 +1,90 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from astropy.wcs import WCS
+from astropy.coordinates import SkyCoord
+
+__all__ = [
+    'WCSGeometry',
+]
+
+
+class WCSGeometry:
+    """Container for WCS object and image shape.
+
+    Parameters
+    ----------
+    wcs : `~astropy.wcs.WCS`
+        WCS projection object
+    shape : tuple
+        Shape of the image
+
+    Examples
+    --------
+    ::
+
+        >>> from hips.utils import WCSGeometry
+        >>> from astropy.coordinates import SkyCoord
+        >>> skycoord = SkyCoord(10, 20, unit="deg")
+        >>> wcs_geometry = WCSGeometry.create(skydir=skycoord, shape=(10, 20), coordsys='CEL', projection='AIT', cdelt=1.0, crpix=1.)
+        >>> wcs_geometry.wcs
+        Number of WCS axes: 2
+        CTYPE : 'RA---AIT'  'DEC--AIT'
+        CRVAL : 10.0  20.0
+        CRPIX : 1.0  1.0
+        PC1_1 PC1_2  : 1.0  0.0
+        PC2_1 PC2_2  : 0.0  1.0
+        CDELT : -1.0  1.0
+        NAXIS : 0  0
+        >>> wcs_geometry.shape
+        (10, 20)
+    """
+
+    def __init__(self, wcs: WCS, shape: tuple) -> None:
+        self.wcs = wcs
+        self.shape = shape
```

Comment:
@adl1995 - how about introducing a `namedtuple` for shape and convert the given shape to a `Shape` here?
Then everywhere else in the code we could use `shape.nx` and `shape.ny`, which is more readable than using `[0]` and `[1]` where one always has to think which is which, no?

```
In [1]: shape = (10, 20)

In [2]: from collections import namedtuple

In [3]: namedtuple('Shape', ['ny', 'nx'])
Out[3]: __main__.Shape

In [4]: Shape = namedtuple('Shape', ['ny', 'nx'])

In [5]: Shape((10, 20))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-44cf43da6338> in <module>()
----> 1 Shape((10, 20))

TypeError: __new__() missing 1 required positional argument: 'nx'

In [6]: Shape(10, 20)
Out[6]: Shape(ny=10, nx=20)

In [7]: Shape(10, 20).nx
Out[7]: 20

In [8]: Shape(10, 20).ny
Out[8]: 10

In [9]: Shape(*shape)
Out[9]: Shape(ny=10, nx=20)
```